Flutter_TDD_Architecture_Course_notes.md

```diff
@@ -277,3 +277,302 @@ class GetRandomNumberTrivia extends UseCase<NumberTrivia, NoParams> {
 And now the **second** test will pass.
 
 ![Second_test_pass](https://user-images.githubusercontent.com/27420533/74590474-aa92e180-5006-11ea-8a88-ef8a9a6311e7.png)
+
+#### 4. Data Layer Overview & Models
+> Video 4: https://www.youtube.com/watch?v=keaTZ9M_U1A&t=1516s
+
+In the `fourth` tutorial start by giving a brief explanation of how the models work.
+
+The models are like entities but with more features in the case of this specific application will deal with making the data conversation in Json to Dart.
+
+After this explanation is necessary to create a file inside the **models** folder with the name `number_trivia_model.dart`.
+And then its test version inside the **test** folder with the name `number_trivia_model_test.dart`.
+
+This will be the code that will be inside the test file.
+
+```dart
+import 'package:clean_architecture_tdd_prep/features/number_trivia/data/models/number_trivia_model.dart';
+import 'package:clean_architecture_tdd_prep/features/number_trivia/domain/entities/number_trivia.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  final tNumberTriviaModel = NumberTriviaModel(number: 1, text: 'Test Text');
+
+  test(
+    'should be a subclass of NumberTrivia entity',
+    () async {
+      // assert
+      expect(tNumberTriviaModel, isA<NumberTrivia>());
+    },
+  );
+}
+```
+The test will **not pass** so we will go to the `number_trivia_model.dart` file and pass some parameters.
+
+```dart
+import 'package:meta/meta.dart';
+
+import '../../domain/entities/number_trivia.dart';
+
+class NumberTriviaModel extends NumberTrivia {
+  NumberTriviaModel({
+    @required String text,
+    @required int number,
+  }) : super(
+          text: text,
+          number: number,
+        );
+}
+```
+Then it is necessary to create inside the test folder a folder called `fixtures` that will contain two files one `trivia.json` and the other `trivia_double.json`.
+
+![fixtures](https://user-images.githubusercontent.com/27420533/75034654-94bd6a80-54a5-11ea-96a2-7c7bd2942f8a.png)
+
+This will be the code for `trivia.json`.
+
+```dart
+{
+  "text": "Test Text",
+  "number": 1,
+  "found": true,
+  "type": "trivia"
+}
+```
+And for `trivia_double.json`.
```

Comment:
```suggestion
And for `trivia_double.json` the only difference is that the **"number"** value is a [double](https://dart.dev/articles/archive/numeric-computation) `1.0` instead of an integer `1`:
```